### PR TITLE
Hapi

### PR DESCRIPTION
**Pull request by RolandDoll**
Completed Hapi implementation. I also have the Prototype ready for PR after this goes through.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        if (f) throw new TypeError("Generator is already executing.");
+        while (_) try {
+            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
+            if (y = 0, t) op = [op[0] & 2, t.value];
+            switch (op[0]) {
+                case 0: case 1: t = op; break;
+                case 4: _.label++; return { value: op[1], done: false };
+                case 5: _.label++; y = op[1]; op = [0]; continue;
+                case 7: op = _.ops.pop(); _.trys.pop(); continue;
+                default:
```
Comment (lawhorkl) [inline index.js]: compiled JS files should not be included in a repo.

File: index.ts [hunk 2]
```diff
@@ -1,3 +1,22 @@
...
+    await server.start();
+    console.log('Server running on %s', server.info.uri);
+};
+
+process.on('unhandledRejection', (err) => {
+
+    console.log(err);
+    process.exit(1);
+});
+
```
Comment (lawhorkl) [inline index.ts]: does this API have a testing route?

**Comment by RolandDoll** [comment]
Compiled JS is now in gitignore and a demonstration route has been added.

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,5 @@
+node_modules/
+
 .DS_Store
 
-node_modules
+index.js
```
Comment (lawhorkl) [inline .gitignore]: We should probably try to find a way to handle complied JS files as we go as we will definitely have more than one in the future... Maybe we should have them output in the `dist` folder and then add the `dist` folder to the gitignore?

**Comment by RolandDoll** [comment]
Dist file has been set up and put in gitignore

**Review by lawhorkl — APPROVED** [review]

**Pull request merged**